public/app/core/services/keybindingSrv.ts

```diff
@@ -193,15 +193,38 @@ export class KeybindingSrv {
       this.bind('x', async () => {
         if (dashboard.meta.focusPanelId) {
           const panel = dashboard.getPanelById(dashboard.meta.focusPanelId);
-          const datasource = await this.datasourceSrv.get(panel.datasource);
-          if (datasource && datasource.supportsExplore) {
+
+          let exploreDatasource = await this.datasourceSrv.get(panel.datasource);
+          let exploreTargets = panel.targets;
+
+          // Mixed datasources need to choose only one datasource
+          if (exploreDatasource.meta.id === 'mixed' && exploreTargets) {
+            // Find first explore datasource among targets
+            let mixedExploreDatasource;
+            for (const t of panel.targets) {
+              if (!mixedExploreDatasource) {
+                const datasource = await this.datasourceSrv.get(t.datasource);
+                if (datasource && datasource.meta.explore) {
+                  mixedExploreDatasource = datasource;
```

Comment:
normally you would have a break here instead of if condition at the top of the loop. I wonder if we can break the code here out into a explore util function under core/utils/explore.ts ? that might make it easier to test. 

The reason for the problems with url change is due to this beind done in an async function so will be outside an angular digest. I would suggest using import store and updateLocation action send dispatch that instead. 

example:

```
store.dispatch(
      updateLocation({
        path: 'explore',
        query: {state: exploreState}
      })
    );
```

Want to migrate way from using location angular service. And the location store -> angular location bridge will make sure it happens inside an angular digest cycle (how angular detects changes)